src/lib/builders/tooltip/types.ts

```diff
@@ -32,7 +32,7 @@ export type CreateTooltipProps = {
 	/**
 	 * Optionally override the default ids we assign to the elements
 	 */
-	ids?: IdObj<TooltipIdParts>;
+	ids?: Partial<IdObj<TooltipIdParts>>;
```

Comment:
You can make the IdObj itself be a partial

Reply:
Tried it. The entire project exploded with type errors everywhere. I tried to fix them, but the calendar builder relies on the ids being required, not partial.

Reply:
You’ll find the original commit in the history. Try it for yourself.

Reply:
Yeah @TGlide the issue with making the `IdObj` a partial is that the type is also used internally within the builders for the `ids` object so it expects all the properties to exist.

Reply:
Yeah, it's probbly not worth the hassle and confusion. Merging!